airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/split_secrets/SecretsHelpers.java

```diff
@@ -408,4 +409,71 @@ protected static SecretCoordinate getCoordinate(
     return new SecretCoordinate(coordinateBase, version);
   }
 
+  public static SecretCoordinateToPayload convertStagingConfigToSecret(final String newSecret,
+                                                                       final ReadOnlySecretPersistence secretReader,
+                                                                       final UUID destinationDefinitionId,
+                                                                       final Supplier<UUID> uuidSupplier,
+                                                                       final @Nullable JsonNode oldStagingConfiguration) {
+    final String oldSecretFullCoordinate =
+        (oldStagingConfiguration != null && oldStagingConfiguration.has(COORDINATE_FIELD)) ? oldStagingConfiguration.get(COORDINATE_FIELD).asText()
+            : null;
+    final SecretCoordinate coordinateForStagingConfig = getCoordinateForStagingConfig(newSecret,
+        secretReader,
+        destinationDefinitionId,
+        uuidSupplier,
+        oldSecretFullCoordinate);
+    return new SecretCoordinateToPayload(coordinateForStagingConfig, newSecret,
+        Jsons.jsonNode(Map.of(COORDINATE_FIELD, coordinateForStagingConfig.getFullCoordinate())));
+  }
+
+  private static SecretCoordinate getCoordinateForStagingConfig(
+                                                                final String newSecret,
+                                                                final ReadOnlySecretPersistence secretReader,
+                                                                final UUID destinationDefinitionId,
+                                                                final Supplier<UUID> uuidSupplier,
+                                                                final @Nullable String oldSecretFullCoordinate) {
+    String coordinateBase = null;
+    Long version = null;
+
+    if (oldSecretFullCoordinate != null) {
+      final var oldCoordinate = SecretCoordinate.fromFullCoordinate(oldSecretFullCoordinate);
+      coordinateBase = oldCoordinate.getCoordinateBase();
+      final var oldSecretValue = secretReader.read(oldCoordinate);
+      if (oldSecretValue.isPresent()) {
+        if (oldSecretValue.get().equals(newSecret)) {
+          version = oldCoordinate.getVersion();
+        } else {
+          version = oldCoordinate.getVersion() + 1;
+        }
+      }
+    }
+
+    if (coordinateBase == null) {
+      // IMPORTANT: format of this cannot be changed without introducing migrations for secrets
+      // persistences
+      coordinateBase = "destination_definition_" + destinationDefinitionId + "_secret_" + uuidSupplier.get();
```

Comment:
This probably shouldn't be tied to the destination definition id. The workspace scoping above was to identify the "owner" of the secret. 

So it could be either `actor_definition` or `staging_config` too.